uaclient/entitlements/tests/test_fips.py

```diff
@@ -927,10 +927,12 @@ def test_fips_does_not_show_enabled_when_fips_updates_is(
 
 class TestFipsEntitlementInstallPackages:
     @mock.patch(M_PATH + "apt.run_apt_command")
+    @mock.patch(M_PATH + "apt.remove_repo_from_apt_auth_file")
     def test_install_packages_fail_if_metapackage_not_installed(
-        self, m_run_apt, entitlement
+        self, m_remove_apt, m_run_apt, entitlement
     ):
         m_run_apt.side_effect = exceptions.UserFacingError("error")
+        m_remove_apt.side_effect = None
         with pytest.raises(exceptions.UserFacingError):
             entitlement.install_packages()
```

Comment:
Lets mock the `_cleanup` method on the entitlement directly. Instead of the `apt` function which is an implementation detail of the `_cleanup` method. Something like this:
```python
    @mock.patch(M_PATH + "apt.run_apt_command")
    def test_install_packages_fail_if_metapackage_not_installed(
        self, m_run_apt, entitlement
    ):
        m_run_apt.side_effect = exceptions.UserFacingError("error")
        with mock.patch.object(
            entitlement,
            "_cleanup",
        ):
            with pytest.raises(exceptions.UserFacingError):
                entitlement.install_packages()
```

Reply:
This seems a better approach. Tested it and works. Updating right now.